Tests/MongoSwiftSyncTests/CrudTests.swift

```diff
@@ -93,20 +93,22 @@ final class CrudTests: MongoSwiftTestCase {
         )
         let runner = try UnifiedTestRunner()
         var skipList: [String: [String]] = [
-            // libmongoc chose not to implement CDRIVER-3630 in anticipation of DRIVERS-1340,
-            // so we cannot pass these tests for now.
-            // TODO: DRIVERS-1340 unskip
-            "unacknowledged-findOneAndUpdate-hint-clientError": ["*"],
-            "unacknowledged-updateMany-hint-clientError": ["*"],
-            "unacknowledged-updateOne-hint-clientError": ["*"],
-            "unacknowledged-deleteOne-hint-clientError": ["*"],
-            "unacknowledged-deleteMany-hint-clientError": ["*"],
-            "unacknowledged-replaceOne-hint-clientError": ["*"],
-            "unacknowledged-findOneAndReplace-hint-clientError": ["*"],
-            "unacknowledged-findOneAndDelete-hint-clientError": ["*"],
-            "unacknowledged-bulkWrite-delete-hint-clientError": ["*"],
-            "unacknowledged-bulkWrite-update-hint-clientError": ["*"],
-            "unacknowledged-bulkWrite-replace-hint-clientError": ["*"]
+            // Skipped because ignored unacknowledged write concern via SWIFT-1304
```

Comment:
To be honest, I don't really understand. This was attached to the Spec documentation:  
"Drivers (like Java, for example) that ignore unacknowledged write concern for findOneAndReplace/Update/Delete helpers should skip the following tests:"

These exact tests were failing, so I figured that the Swift driver ignores unacknowledged write concerns for FindOneAndX operations. Didn't want to create too lengthy of a comment since its not a docstring so I kept it small. Thoughts? Maybe something like "Skipped because driver ignores unacknowledged write concerns for FindOne, SWIFT-1304"

Reply:
so we do respect unacknowledged write concerns for those operations, however it looks like these are failing due to a small issue in the unified test runner. 

for some context, an unacknowledged write concern is something the user can use to indicate that they don't want the driver to wait for the server to "acknowledge" that it actually received the write and stored the data.
the CRUD spec gives drivers a couple options for what to return from various write methods like `insertOne` when the write is unacknowledged: one choice is `nil` (or the language equivalent), and the other is to return an object that has a property `acknowledged: false`. for Swift, we went with `nil` in our API.
however, for the purpose of making assertions about the responses to unacknowledged writes in the unified test runner, we introduced a special placeholder object `UnifiedOperationResult.unacknowledgedWrite` [here](https://github.com/mongodb/mongo-swift-driver/blob/3eb642c786eeaacf236b48fb9a2b9ad1e5561c7f/Tests/MongoSwiftSyncTests/UnifiedTestRunner/UnifiedOperation.swift#L22-L25) that we return from test operations when the write is unacknowledged. at least at the time we added that, there were some unified tests that expected drivers to do the `acknowledged: false` form rather than returning nil, so using the explicit result allowed us to pass those tests,

however, the small issue is that the `acknowledged: false` placeholder response doesn't actually make any sense to return from the `findOneAndX` operations, because when it comes to those methods, the CRUD spec does not have drivers return `acknowledged: false` -- it expects drivers return BSON null or nil. (which is a bit confusing since it can be conflated with the "find" part of the operation not finding a result, and is perhaps why drivers such as Java chose not to support doing these type of writes unacknowledged.)

so I think the right fix here is to run these tests after all, but to update `UnifiedFindOneAndUpdate`, `UnifiedFindOneAndDelete`, and `UnifiedFindOneAndReplace` to return `.none`, (which will match the expected test result) rather than `.unacknowledgedWrite`.

Reply:
This makes a lot of sense. Thanks for the explanation, I just assumed that since the specs said the tests might fail that its bc the driver doesnt support it. Made these changes, which clears out the skip list. When looking at the block of code below that checks compiler version and adds a test to the skip list, omitting that whole section did not cause any failures and so I removed everything related to the `skipList`

Reply:
> when looking at the block of code below that checks compiler version
just FYI that was checking server version, not compiler version 🙂 but yeah, we were skipping that due to a bug which has now been fixed. SWIFT-1359 was tracking unskipping that test, so once you merge this PR, can you please close that ticket out too, and leave a comment on it saying it was fixed via this PR (and link to this PR)?